google/ggrc-core
# GGRC-5333  Implement basic framework for issuetracker integration

# Dependencies

This PR is `on hold` until the dependencies are merged:

- [x] #8075 

# Issue description
The idea of this PR  is implementation of some functionality for comfortable further implementation of issue object integration. Currently those functionality hasn't been used in application at all and it doesn't affect existing functionality (except a small code refactoring). This functionality will be used in further PRs.
 
This PR contains next changes:
* Common event handlers for issue tracker has been added (see _Solution description_ section for more details);
* Small refactoring (moving some functions and tests for assessment integration in separate modules);
* Issue tracker query builder has been added (see _Solution description_ section for more details);
* Unit tests for query builder has been added.

# Steps to test the changes

Currently it's hard to check those functionality with clicking the app (because it doesn't affect other functionality). But it will be easy to check those functionality with GGRC-5293, GGRC-5295

# Solution description

I have implemented common event handlers for integration functionality. The main idea of this changes is containing event handlers in declarative notation. You can set `create`, `update` and `delete` handlers for models in `ISSUE_TRACKER_HANDLERS`.
There are common handlers for create, update and delete events wich will call concrete handlers for concrete model.

Also this PR contains `BaseIssueTrackerQueryBuilder` and `IssueQueryBuilder`. The main idea of those classes is containing business logic in separate class for which we can easily write unit-tests. In case of changing business we will need rewrite only query builder and handlers will not require any changes.

# Sanity checklist




# PR Review checklist
